segment.go

```diff
@@ -168,3 +168,14 @@ type DocVisitState interface {
 type StatsReporter interface {
 	ReportBytesWritten(bytesWritten uint64)
 }
+
+type FieldStatsReporter interface {
+	UpdateFieldStats(FieldStats)
+}
+
+type FieldStats interface {
+	AddStat(statName, fieldName string, value uint64)
+	AggregateStats(stats FieldStats)
+	AppendStat(statName, fieldName string, value uint64)
```

Comment:
This and the previous API should be consolidated into 1 API, there's no need for two here.

Reply:
FieldStats is an interface implemented by a structure in bleve which is used to store the stats while FieldStatsReporter is implemented by a SegmentBase. So, I don't see how I can merge the two apis into one.

Reply:
Consolidating Aggregate and Append is what I was referring to :)